homeassistant/components/smarty/switch.py

```diff
@@ -0,0 +1,93 @@
+"""Support for Salda Smarty XP/XV Ventilation Unit Sensors."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+import logging
+from typing import Any
+
+from pysmarty2 import Smarty
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .coordinator import SmartyConfigEntry, SmartyCoordinator
+from .entity import SmartyEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class SmartySwitchDescription(SwitchEntityDescription):
+    """Class describing Smarty switch."""
+
+    is_on_fn: Callable[[Smarty], bool]
+    turn_on_fn: Callable[[Smarty], bool | None]
+    turn_off_fn: Callable[[Smarty], bool | None]
+
+
+ENTITIES: tuple[SmartySwitchDescription, ...] = (
+    SmartySwitchDescription(
+        key="boost",
+        translation_key="boost",
+        is_on_fn=lambda smarty: bool(smarty.boost),
+        turn_on_fn=lambda smarty: smarty.enable_boost(),
+        turn_off_fn=lambda smarty: smarty.disable_boost(),
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: SmartyConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Smarty Switch Platform."""
+
+    coordinator = entry.runtime_data
+
+    async_add_entities(
+        SmartySwitch(coordinator, description) for description in ENTITIES
+    )
+
+
+class SmartySwitch(SmartyEntity, SwitchEntity):
+    """Representation of a Smarty Switch."""
+
+    entity_description: SmartySwitchDescription
+
+    def __init__(
+        self,
+        coordinator: SmartyCoordinator,
+        entity_description: SmartySwitchDescription,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._is_on = False
```

Comment:
There is no use for `self._is_on` so it can be removed

Reply:
Done